### PR TITLE
x-pack/filebeat/input/netflow: allow v9 template sharing

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This adds an option to allow users to specify sharing of v9 and ipfix templates within a session without regard to the template/flow origin.

## Why is it important?

This behaviour is the default option in logstash and the current filebeat behaviour is blocking users migrating from logstash.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Review by andrewkroh — COMMENTED** [review]
There is an asciidoc file that should be updated with the new option too.

https://www.elastic.co/guide/en/beats/filebeat/master/filebeat-input-netflow.html#_configuration_options_11

I think we should include some kind of disclaimer that enabling this goes against the IPFIX and Netflow RFCs and can result in the wrong template being applied under certain conditions...

**Comment by efd6** [comment]
The language in the RFC is not particularly strong wrt this. I'll like to the sections that get close to it in the final commit message (linked in the commit that adds the docs).

File: x-pack/filebeat/input/netflow/decoder/config/config.go [hunk 1]
```diff
@@ -81,6 +83,14 @@ func (c *Config) WithCustomFields(dicts ...fields.FieldDict) *Config {
 	return c
 }
 
+// WithSharedTemplates allows to toggle the sharing of templates within
+// a v9 neflow session. If it is not enabled, the source address must match
```
Comment (andrewkroh) [inline x-pack/filebeat/input/netflow/decoder/config/config.go:87]:
```suggestion
// a v9 netflow or ipfix session. If it is not enabled, the source address must match
```

**Review by andrewkroh — APPROVED** [review]
LGTM. Just a minor doc suggestion and typo.

**Pull request merged**